compose/lite/docker-compose.yml

```diff
@@ -0,0 +1,89 @@
+version: '3.7'
+
+networks:
+  net:
+    driver: bridge
+
+services:
+  authelia:
+    image: authelia/authelia
+    container_name: authelia
+    volumes:
+      - ./authelia:/var/lib/authelia
+      - ./configuration.yml:/etc/authelia/configuration.yml:ro
+      - ./users_database.yml:/etc/authelia/users_database.yml:ro
+    networks:
+      - net
+    labels:
+      - 'traefik.enable=true'
+      - 'traefik.http.routers.authelia.rule=Host(`auth.example.com`)'
+      - 'traefik.http.routers.authelia.entrypoints=https'
+      - 'traefik.http.routers.authelia.tls=true'
+      - 'traefik.http.routers.authelia.tls.certresolver=letsencrypt'
```

Comment:
Will this line make traefik request a cert to letsencrypt? I'd rather provide a self-signed cert to not rely on the fact that users have a DNS already set up (This because I think it's a barrier for testing Authelia, the minimum should be required to bootstrap the minimal setup).

Reply:
That would mean that the user has to modify their hosts file like part of the bootstrap process, I don't think that's just as simple as "run this docker-compose".
Whereas including the LetsEncrypt configuration means that by virtue of exposing port 80 in the compose stack it will fetch the required certificates.

I would also say it should be a safe assumption to make that if someone plans to test this that they likely already have some domains setup and likely already Traefik or nginx too.

Reply:
Do you mean that the certificate will be generated even if the machine is not publicly available?

I think it won't. If traefik leverages the certbot with http method, it means that let's encrypts needs to complete the challenge with public access to the machine. In that case it means 2 things: opening the machine to the public and creating a DNS entry. It might be a barrier for people starting from scratch on a laptop. The /etc/host method has only one simple prerequisite instead of two that's why I like it better. If we go that way we might keep the cert generation options but comment them.

What do you think?

Reply:
After second thought, I think we have 2 different objectives. You want to expose the maximum features in the minimal snippet so that people can start from a seed and adapt to their context but they MUST adapt (DNS and public exposition, mail server, etc...)

On my side, I'm currently working on the getting started and I'm trying to provide the easiest way to test authelia locally so the less pre-requisites the better. Now that I'm aware of this, I think your PR solves your objective and I will work on mine independently based on your docker-compose. We'll see if we can find a way to factorize afterwards if possible.

Reply:
> Do you mean that the certificate will be generated even if the machine is not publicly available?

The only way to make this work would be to use DNS based challenges as opposed to HTTP or TLS-SNI.

Having said that to make the webserver work as it would be intended via internet access the service would need to be exposed anyway, so I don't think we're really solving much by moving to DNS based challenges alone.